In [69]:
import pandas as pd

In [70]:
dataset = pd.read_csv("preprocessed_placement.csv")

In [71]:
dataset

ssc_p  hsc_p  degree_p  etest_p  mba_p         salary gender    ssc_b  \
0    67.00  91.00     58.00     55.0  58.80  270000.000000      M   Others   
1    79.33  78.33     77.48     86.5  66.28  200000.000000      M  Central   
2    65.00  68.00     64.00     75.0  57.80  250000.000000      M  Central   
3    56.00  52.00     52.00     66.0  59.43  288655.405405      M  Central   
4    85.80  73.60     73.30     96.8  55.50  425000.000000      M  Central   
..     ...    ...       ...      ...    ...            ...    ...      ...   
210  80.60  82.00     77.60     91.0  74.49  400000.000000      M   Others   
211  58.00  60.00     72.00     74.0  53.62  275000.000000      M   Others   
212  67.00  67.00     73.00     59.0  69.72  295000.000000      M   Others   
213  74.00  66.00     58.00     70.0  60.23  204000.000000      F   Others   
214  62.00  58.00     53.00     89.0  60.22  288655.405405      M  Central   

       hsc_b     hsc_s   degree_t workex specialisation      status  
0     Others  Commerce   Sci&Tech     No         Mkt&HR      Placed  
1     Others   Science   Sci&Tech    Yes        Mkt&Fin      Placed  
2    Central      Arts  Comm&Mgmt     No        Mkt&Fin      Placed  
3    Central   Science   Sci&Tech     No         Mkt&HR  Not Placed  
4    Central  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
..       ...       ...        ...    ...            ...         ...  
210   Others  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
211   Others   Science   Sci&Tech     No        Mkt&Fin      Placed  
212   Others  Commerce  Comm&Mgmt    Yes        Mkt&Fin      Placed  
213   Others  Commerce  Comm&Mgmt     No         Mkt&HR      Placed  
214   Others   Science  Comm&Mgmt     No         Mkt&HR  Not Placed  

[215 rows x 14 columns]

In [72]:
dataset[dataset["gender"]=='M']

ssc_p  hsc_p  degree_p  etest_p  mba_p         salary gender    ssc_b  \
0    67.00  91.00     58.00     55.0  58.80  270000.000000      M   Others   
1    79.33  78.33     77.48     86.5  66.28  200000.000000      M  Central   
2    65.00  68.00     64.00     75.0  57.80  250000.000000      M  Central   
3    56.00  52.00     52.00     66.0  59.43  288655.405405      M  Central   
4    85.80  73.60     73.30     96.8  55.50  425000.000000      M  Central   
..     ...    ...       ...      ...    ...            ...    ...      ...   
209  62.00  72.00     65.00     67.0  56.49  216000.000000      M  Central   
210  80.60  82.00     77.60     91.0  74.49  400000.000000      M   Others   
211  58.00  60.00     72.00     74.0  53.62  275000.000000      M   Others   
212  67.00  67.00     73.00     59.0  69.72  295000.000000      M   Others   
214  62.00  58.00     53.00     89.0  60.22  288655.405405      M  Central   

       hsc_b     hsc_s   degree_t workex specialisation      status  
0     Others  Commerce   Sci&Tech     No         Mkt&HR      Placed  
1     Others   Science   Sci&Tech    Yes        Mkt&Fin      Placed  
2    Central      Arts  Comm&Mgmt     No        Mkt&Fin      Placed  
3    Central   Science   Sci&Tech     No         Mkt&HR  Not Placed  
4    Central  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
..       ...       ...        ...    ...            ...         ...  
209  Central  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
210   Others  Commerce  Comm&Mgmt     No        Mkt&Fin      Placed  
211   Others   Science   Sci&Tech     No        Mkt&Fin      Placed  
212   Others  Commerce  Comm&Mgmt    Yes        Mkt&Fin      Placed  
214   Others   Science  Comm&Mgmt     No         Mkt&HR  Not Placed  

[139 rows x 14 columns]

# Unpaired t-test
# Different group (male, female) but same condition (salary)

In [73]:
from scipy.stats import ttest_ind
# As we are using preprocessed data, I commented the below line
#dataset = dataset.dropna()

In [74]:
male_salary = dataset[dataset["gender"]=='M']["salary"]
female_salary = dataset[dataset["gender"]=='F']["salary"]

In [75]:
ttest_ind(male_salary, female_salary)

TtestResult(statistic=1.9001627265152634, pvalue=0.05876239427122641, df=213.0)

# Paired t-test
# Same group (male) but different conditions (ssc_p, hsc_p)

In [76]:
from scipy.stats import ttest_rel

In [77]:
male_ssc_p = dataset[dataset["gender"]=='M']["ssc_p"]
male_hsc_p = dataset[dataset["gender"]=='M']["hsc_p"]

In [78]:
ttest_rel(male_ssc_p, male_hsc_p)

TtestResult(statistic=0.6852881050898076, pvalue=0.49431164600731514, df=138)

# Assignment - I
# Compare salary of central and others hsc_b

In [79]:
central_hsc_b_salary = dataset[dataset["hsc_b"]=='Central']["salary"]
others_hsc_b_salary = dataset[dataset["hsc_b"]=='Others']["salary"]

In [80]:
ttest_ind(central_hsc_b_salary, others_hsc_b_salary)

TtestResult(statistic=0.09117578929975474, pvalue=0.9274385986418501, df=213.0)

# Here p values is 0.92 > 0.05 so the difference is high

# Assignment -II
# Compare mba_p and  etest_p of commerce in hsc_s

In [81]:
commerce_group_mba_p = dataset[dataset["hsc_s"]=='Commerce']['mba_p']
commerce_group_etest_p = dataset[dataset["hsc_s"]=='Commerce']['etest_p']

In [82]:
ttest_rel(commerce_group_mba_p, commerce_group_etest_p)

TtestResult(statistic=-7.868552092606871, pvalue=2.462926468454984e-12, df=112)

# Here p values is 2.4 > 0.05 so the difference is high

# Assignment -III

# Compare mba commerce and Science group

In [83]:
commerce_mba_p = dataset[dataset["hsc_s"]=='Commerce']['mba_p']
science_mba_p = dataset[dataset["hsc_s"]=='Science']['mba_p']

In [84]:
ttest_ind(commerce_mba_p, science_mba_p)

TtestResult(statistic=-0.7331285580404581, pvalue=0.46432995253854314, df=202.0)

# Here p values is 0.46 > 0.05 so the difference is high